spec/controllers/rents_controller_spec.rb

```diff
@@ -0,0 +1,170 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe RentsController do
+  include_context 'Authenticated User'
+
+  describe 'GET #create' do
+    context 'When creating a rent for an user' do
+      let!(:user) { create(:user) }
+      let!(:book) { create(:book) }
+      let!(:rent) { attributes_for(:rent, book_id: book.id) }
+      before do
+        post :create, params: { user_id: user.id, rent: rent }
+      end
+
+      it 'returns http success' do
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'responds with the created rent' do
+        expect(response_body).to have_key('loan')
+        expect(response_body).to have_key('restitution')
+        expect(response_body).to have_key('user')
+        expect(response_body).to have_key('book')
+      end
+    end
+
+    context 'When given a non existent user ID' do
+      let!(:book) { create(:book) }
+      let!(:rent) { attributes_for(:rent, book_id: book.id) }
+      before do
+        post :create, params: { user_id: 1, rent: rent }
+      end
+
+      it 'returns http status 404' do
+        expect(response).to have_http_status(:not_found)
+      end
+
+      it 'responds with an error message' do
+        expect(response_body).to have_key('error')
+      end
+    end
+  end
+
+  describe 'GET #index' do
+    context 'When fetching all the rents of a book' do
+      let!(:book) { create(:book) }
+      let!(:rents) do
+        today = Time.zone.today
```

Comment:
You made this because you are validating rents overlapping??

Reply:
Yes, if the dates aren't checked to be differents, the rent is not going to be created because it is used the same date